### PR TITLE
Fixes issues #10367 #10302 #10346

### PR DESCRIPTION
**Pull request by erangatl**
Fixes: https://github.com/wso2/product-apim/issues/10367, Fixes: https://github.com/wso2/product-apim/issues/10302, Fixes: https://github.com/wso2/product-apim/issues/10346

### CODE REVIEW

**Review by malinthaprasan — APPROVED** [review]

**Review by Krishanx92 — APPROVED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentAdd.jsx [hunk 1]
```diff
@@ -96,8 +96,9 @@ class CommentAdd extends React.Component {
      * @memberof CommentAdd
      */
     handleClickCancel() {
-        const { toggleShowReply } = this.props;
-        toggleShowReply();
+        this.setState({ content: '' });
+        const { handleShowReply } = this.props;
+        handleShowReply(-1);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentAdd.jsx:101]: why do we need to pass a `-1` to a handler ?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentOptions.jsx [hunk 2]
```diff
@@ -150,13 +150,14 @@ class CommentOptions extends React.Component {
         const canModify = comment.createdBy === AuthManager.getUser().name && comment.entryPoint === 'APIPublisher';
         return (
             <Grid container spacing={2} className={classes.verticalSpace} key={comment.id}>
-                {comment.replyTo == null && [
+                {comment.parentCommentId == null && [
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentOptions.jsx:153]: should be `===`

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Comments/CommentOptions.jsx [hunk 3]
```diff
@@ -166,13 +167,14 @@ class CommentOptions extends React.Component {
 
                     ]}
 
-                {comment.replyTo == null && [
+                {comment.parentCommentId == null && [
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Comments/CommentOptions.jsx:170]: Should be `===`

Comment (erangatl) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Comments/CommentOptions.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentOptions.jsx (hunk 2 above)
Comment (erangatl) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentOptions.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentAdd.jsx (hunk 1 above)
Comment (erangatl) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/CommentAdd.jsx]: Fixed

**Pull request merged**